### DISCUSSION TITLE
DNS Cloaking and Forwarding Based on WiFi SSID

### DISCUSSION THREAD

**Discussion by 9p4** [DNSCrypt/dnscrypt-proxy#1913] [Ideas]
I am using dnscrypt-proxy in an environment that uses split horizon DNS. Internal sites will resolve but only using the internal DNS server. Using dnscrypt-proxy means that I will have to manually change the rules every time I switch connections (inside and outside this environment). Is it possible to apply rules based on the WiFi SSID I am connected to?

**Comment by lifenjoiner** [comment]
Write forwarding rules for the LAN domains should work: https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Forwarding
Or you can write a script that restart `dnscrypt-proxy` with a specified config file for the specified environment. `dnscrypt-proxy` doesn't have the feature of config file reloading.

**Reply by 9p4** [reply]
Unfortunately, the LAN domains are companyname.com, which are the same externally and internally.

**Comment by ianbashford** [comment]
Restarting with a different config file on change of Lan is a good idea. 
otherwise dnscrypt-proxy has a small footprint. You could run two instances with different configurations and use the Lan/WiFi DNS settings to use the appropriate instance.

**Reply by 9p4** [reply]
This seems more like a workaround than a solution imo. I know some Go, so I'll see if I can put together a PR.
